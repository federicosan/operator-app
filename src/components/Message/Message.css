.Message__client {
  align-self: flex-end;
  background: #ffffff;
  border-radius: 10px 10px 0 10px;
  color: #000;
  float: left;
  font-size: 12px;
  letter-spacing: 1px;
  margin-top: 6px;
  margin-right: 6px;
  max-width: 45%;
  padding: 10px;
  text-align: left;
}

.Message__operator {
  align-self: flex-start;
  background: rgb(230, 126, 34);
  border-radius: 2px;
  color: white;
  font-size: 12px;
  letter-spacing: 1px;
  padding: 0.5rem;
  margin: 0.5rem;
  max-width: 45%;
}
