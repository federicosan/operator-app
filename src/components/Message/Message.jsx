/**
  * @summary: Displays a single message in the Message List
*/

import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import './Message.css';

const Message = props => {
  const msgClass = () => (
    props.type === 'operator' ? 'Message__operator' : 'Message__client'
  );

  return <div className={msgClass()}>{props.children}</div>
};



export default Message


Message.propTypes = {
  children: PropTypes.node.isRequired,
};
