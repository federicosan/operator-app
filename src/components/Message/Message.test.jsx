
import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message.jsx';

/*
const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({})),
};
*/


describe('MessageList', () => {
  it('matches snapshot', () => {
    const component = shallow(<Message type="operator">test message</Message>);
    expect(component).toMatchSnapshot();
  });
});
