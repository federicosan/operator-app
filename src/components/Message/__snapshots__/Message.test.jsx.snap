exports[`MessageList matches snapshot 1`] = `
<div
  className="Message__operator">
  test message
</div>
`;
